### ISSUE TITLE
Stuck @ 99%

### ISSUE THREAD

**Issue by HC1Gunner** [Mmtrx/FS22_ChaffMission#8]
Running the latest version of Better Contracts ([v1.2.8.5a), but forage mission says completed in text in the upper right, but under contracts they all get stuck at 99% and won't show as completed. If you unload any more chaff or forage, it just is seen as sold, and is added to your account.

**Comment by Mmtrx** [comment]
Hi,
can you upload your log file please? Which map / single or MP game?

**Comment by HC1Gunner** [comment]
I'm playing two maps at the same time, Eureka and Adelshoven, they both had
the same issue. Next time I have a forage contract come up, I will send my
log.

**Comment by Gordon-Dry** [comment]
Do you have the both map zip files in the mod folder at the same time by chance?
If so; just only have one to avoid mod issues.

**Comment by HC1Gunner** [comment]
I don't use Giants mod folder, all my maps run from their own map folders
with their mods in those folders, there can be no cross-talk.
What I did notice last night with the chaff mission mod running, and any
contract that has to do with harvesting, it says completed (text) in the
HUD, but under contracts they are all stuck at 99%. Since the game sees
them at 99% the field is now locked, and you will get no further contracts
for it, and the field will remain in the harvest state.
I had to go into savegames and edit the mission xml to show them as
completed.

**Comment by silentille** [comment]
We have the same issue at map HoF 2022 in Multiplayer. Singleplayer not tested. Logfile is attached
[log.txt](https://github.com/Mmtrx/FS22_ChaffMission/files/13064465/log.txt)

**Comment by HC1Gunner** [comment]
I playing nothing but SP, I don't play MP at all.

**Comment by HC1Gunner** [comment]
I supposed my next test would be to remove the Better Contracts mod, and
run just the Chaff missions and see if it messes up the base-game
contracts. Also start a new game with no other mods running except for
Forage Contracts and see how it behaves on its own, then add the latest
version of Better Contracts.

**Comment by HC1Gunner** [comment]
I tried a couple of other maps with just (4) mods running...Better contracts, Chaff missions, Easy Dev, and speed Ctrl, and had no issues. When each field was finished, completion showed @ 80%, delivery to the sell point completed the contract successfully.
Which probably means, that the combination of better contracts and chaff missions have a mod conflict with one of my other mods. Since I have used better contracts since it was released and never had an issue, it probably chaff missions that has the conflict.
I will copy over mods (5 at a time) from one of the maps I was having issues on and see if I can find the culprit.

**Comment by silentille** [comment]
Thanks for your experiments, HC1Gunner. I will attach a list of the mods we are using. Maybe you can  cross-check with your mods. The mod causing the conflict could be part of both mine and your mods.

Contracts worked fine for me on MP too when using Better Contracts + attached mods only.

[modlist.txt](https://github.com/Mmtrx/FS22_ChaffMission/files/13175842/modlist.txt)

**Comment by HC1Gunner** [comment]
Got your mod list, I take these are the mods you're using and are having no issues or conflicts while running better contracts and forage contracts. A quick look and I'm running several of the same mods, so this will save some time during testing.

**Comment by silentille** [comment]
The other way around: this is the modlist which we are facing issues with not completable contracts with.

**Comment by Mmtrx** [comment]
Maybe it's not BC or ChaffMission mods. In @HC1Gunner 's log are some other errors / callstacks, e.g.
```
2023-10-20 21:55 AIVehicleExtension:showWarning: self == nil ( Feld Erkennung läuft (0.568 ha) / 500 )
2023-10-20 21:55 LUA call stack:
  =C://gameservers//s1465577//ls2022//gameProfile/mods/FS22_AIVehicleExtension/mogliHud.lua (59) : 2023-10-20 21:55 AIVehicleExtension:showWarning: self == nil ( Feld Erkennung läuft (0.568 ha) / 500 )
```
Also, CP has problems to determine field boundaries of fields 20, 41, 45, 63, 73, 128, 138. Does subject harvest completion problem happen to occur on one of those fields? 

I have seen similar issues to this, when fields are not properly defined in mod maps. So that the base game cannot correctly determine when a field is completely harvested. 

@HC1Gunner, when you say " forage mission says completed in text in the upper right", you probably mean it says s.th. like "completely delivered for mission...". 
To complete a harvest / chaff mission, the game needs
- complete delivery of the expected amount to the specified sellpoint (-> 20% total completion) AND
- completely worked field (-> 80% total completion)

A good test would be to load the mod map you had the problem on, just BC, EDC, and ChaffMissions. And then generate a chaff mission on one of the fields where the problem occurred. You can generate a chaff mission by advancing the fruit to growth state one below harvest-ready (by EDC). Then say `"chGenerateFieldMission <fieldno.>"` on the console.

**Comment by HC1Gunner** [comment]
Its not just chaff missions, any contract that has to do with harvesting a crop, wheat, corn, etc, they all become stuck at 99%.
I took CP out the equation by manually harvesting fields myself, more than likely its a mod conflict. Even thought FS supports mods, the way Giants has the Sim setup, it not really a mod friendly game, there are way too may leeway's as to what modders can do and get away with that cause a lot of the problems.

**Comment by HC1Gunner** [comment]
While trying a new map, Land of Italy, I found that FS22_ChaffMission will causing the map not to load, locking up at 85%. It produced the below error, not sure if you want a new issue started or what.

550:2023-10-31 17:41 Error: Running LUA method 'loadSharedI3DFileAsyncFinished'.

**Comment by Mmtrx** [comment]
Hi @HC1Gunner, can you post the complete log.txt please?

**Comment by HC1Gunner** [comment]
I was not running the Forage contract mod as the map would not run.  The very first harvest mission I did had the same issue, delivered all the grain to the sell point, only for the mission to be stuck at 99%, I had the same result for (2) more contracts to harvest. If you unloaded any more grain, in the hUD you be shown as selling the crop for a profit, yet in the menu/contracts there all remained stuck at 99%.

Solution, I removed FS22_BetterContracts 1.2.8.5a, and as soon as the map loaded, went into contracts menu, and now they are all shown as completed. Its not just Forage Contracts that has issues, but also Better Contracts, at least on Land of Italy. I'm going to go back to and older version of better contracts and see what happens on the next harvest contract I do.

This is the log of when harvest contracts were stuck @ 99%
[log 99% (1).txt](https://github.com/Mmtrx/FS22_ChaffMission/files/13255785/log.99.1.txt)

**Comment by HC1Gunner** [comment]
I put Better Contracts v1.2.6.2 in, and all my harvest contracts completed.

**Comment by Mmtrx** [comment]
Telling from your log, otgher mods have problems:
```
  =G:/FS22 Mod Maps/Land Of Italy v1.0/FS22_InsideCameraZoom/main.lua (11) : getXMLFloat
  dataS/scripts/vehicles/specializations/Enterable.lua (606) : loadFromXML
  dataS/scripts/vehicles/specializations/Enterable.lua (245) : loadCamerasFromXML
  dataS/scripts/specialization/SpecializationUtil.lua (23)
  dataS/scripts/vehicles/Vehicle.lua (993) : raiseEvent
  =G:/FS22 Mod Maps/Land Of Italy v1.0/FS22_fixAttacherJointRot/fixAttacherJointRot.lua (4) : oldfunc
  dataS/scripts/i3d/I3DManager.lua (134) : asyncCallbackFunction
...
2023-11-03 06:24 G:/FS22 Mod Maps/Land Of Italy v1.0/FS22_Raschiatore/RaschiatoreSP2.i3d (1.21 ms)
2023-11-03 06:24   Error: Real Dirt Color : Error loading body parts for EDOMOD SCRAPER
...
2023-11-03 06:24 Error: Missing dds file 'G:/FS22 Mod Maps/edit mappa/LED_Lightbars_22_52_prefab/texture/LED_lightramp_22_52_normal.dds'
2023-11-03 06:24 Error: Can't load resource 'G:/FS22 Mod Maps/edit mappa/LED_Lightbars_22_52_prefab/texture/LED_lightramp_22_52_normal.dds'.
...
2023-11-03 06:24 G:/FS22 Mod Maps/Land Of Italy v1.0/fs22_newholland100_90/NH100.i3d (91.27 ms)
2023-11-03 06:24   Error: Failed to find child 0 from node beaconLights, only 0 children given
...
2023-11-03 06:24 Warning (script): 'setColorGradingSettings': Argument 1 has wrong type. Expected: String. Actual: Nil 
...
2023-11-03 06:31 Error: Running LUA method 'update'.
G:/FS22 Mod Maps/Land Of Italy v1.0/FS22_AdvancedFarmManager/gui/AIMGuiVehicleFrame.lua:491: attempt to call method 'getConsumerFillUnitIndex' (a nil value)
```
I suggest you remove following mods
- FS22_InsideCameraZoom
- FS22_fixAttacherJointRot
- FS22_Raschiatore
- fs22_newholland100_90
- FS22_AdvancedFarmManager

and please turn on debug mode - on the BC settings menu page.

**Comment by Gordon-Dry** [comment]
I got no issues with AdvancedFarmManager - but I don't use any of the other 4 mods.

**Comment by HC1Gunner** [comment]
Jus so you know, going back to the previous version of better contracts fixed the issued, now to add forage contract back and see if it will work with your older version. If I remember correctly it won't that's what caused me to update better contracts to v1.2.8.5a.

I never had any kind of conflict with Inside camera zoom, I been running it forever, on all my maps.
Sames goes for fixAttacherJointRot,, and FS22_Raschiatore is a required mod for Land of Italy, author has (3) of them, map won't load without them. I will try to do some more testing tonight. Hurt my back and so for now, less time spent at my PC.
Advanced Farm Manager has an update that I missed, so that may fix any issues it might be causing.

**Comment by HC1Gunner** [comment]
I finally got around to doing this, as the author of the map seems to have added very few contracts, there are contracts for harvest and sow, that is about it, nothing for cultivate, etc. Even my mod for generation roller missions has added no contracts. I was hope chaff mission would add some, first I need to get the game to load.
I used the previous version of better contracts, turn on debug, then add the latest v1.2.8.5a, game locks on load @ 85%.

Ran better contracts v1.2.8.5a on the new Zielonka map has the same issue, contracts are all stuck at 99%. Went back to previous version, problem is gone.

[log.txt](https://github.com/Mmtrx/FS22_ChaffMission/files/13288854/log.txt)
